### PR TITLE
[IOTDB-1173] auto-create schema improvement for createMultiTimeSeriesPlan

### PR DESCRIPTION
**Pull request by LebronAl**
see [1173](https://issues.apache.org/jira/browse/IOTDB-1173).

### CODE REVIEW

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java [hunk 1]
```diff
@@ -426,58 +428,77 @@ public synchronized boolean containsKey(PartialPath key) {
...
-        && !(plan instanceof InsertRowsPlan)) {
-      // InsertMultiTabletPlan and InsertRowsPlan have multiple devices, and other types of
-      // InsertPlan have only one device.
-      deviceIds = Collections.singletonList(((InsertPlan) plan).getDeviceId());
+    List<PartialPath> storageGroups = new ArrayList<>();
+    // for InsertPlan, try to just use deviceIds to get related storage groups because there's no
+    // need to call getPaths to concat deviceId and sensor as they will gain same result.
+    // for CreateTimeSeriesPlan, use getPath() to get timeseries to get related storage group
+    // for CreateTimeMultiSeriesPlan, use getPaths() to get all timeseries to get related storage
+    // groups
```
Comment (neuyilan) [inline cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:440]: CreateTimeMultiSeriesPlan -> CreateMultiTimeSeriesPlan

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java [hunk 2]
```diff
@@ -426,58 +428,77 @@ public synchronized boolean containsKey(PartialPath key) {
...
-    if (plan instanceof InsertPlan
-        && !(plan instanceof InsertMultiTabletPlan)
-        && !(plan instanceof InsertRowsPlan)) {
-      // InsertMultiTabletPlan and InsertRowsPlan have multiple devices, and other types of
-      // InsertPlan have only one device.
-      deviceIds = Collections.singletonList(((InsertPlan) plan).getDeviceId());
+    List<PartialPath> storageGroups = new ArrayList<>();
+    // for InsertPlan, try to just use deviceIds to get related storage groups because there's no
+    // need to call getPaths to concat deviceId and sensor as they will gain same result.
+    // for CreateTimeSeriesPlan, use getPath() to get timeseries to get related storage group
```
Comment (neuyilan) [inline cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:438]:
```suggestion
    // for CreateTimeSeriesPlan, use getPath() to get timeseries to get related storage group,
```

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java [hunk 3]
```diff
@@ -500,24 +521,23 @@ private void verifyCreatedSgSuccess(List<PartialPath> deviceIds, PhysicalPlan ph
...
+      TSStatus setStorageGroupResult =
+          metaGroupMember.processNonPartitionedMetaPlan(setStorageGroupPlan);
+      if (setStorageGroupResult.getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()
+          && setStorageGroupResult.getCode()
+              != TSStatusCode.PATH_ALREADY_EXIST_ERROR.getStatusCode()) {
+        throw new MetadataException(
+            String.format(
+                "Status Code: %d, failed to set storage group %s",
+                setStorageGroupResult.getCode(), storageGroup));
+      }
```
Comment (neuyilan) [inline cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:540]: Hi, do we need to allow partial set storage group success? @LebronAl @mychaow @jt2594838 
In the current implementation, if one of the storage groups is created failure, the remaining storage group will not be created.

**Review by neuyilan — COMMENTED** [review]
Good Job!

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java (hunk 3 above)
Comment (mychaow) [reply cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:540]: allow partial set storage group success is better.

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java (hunk 1 above)
Comment (LebronAl) [reply cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:440]: OK

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java (hunk 2 above)
Comment (LebronAl) [reply cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:438]: OK

File: cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java (hunk 3 above)
Comment (LebronAl) [reply cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java:540]: No matter how many storage groups are created, client will get an `NO_STORAGE_GROUP` TSStatus as long as there exists a failed setStorageGroup operation, which means the client need to perform a retry, and the failed setStorageGroup operation will be executed in this client's retry request. As this situation is relatively rare, so current implemention is ok for me.

**Review by mychaow — APPROVED** [review]
LGTM

**Pull request merged**